File: clients/algoliasearch-client-java-2/algoliasearch-core/com/algolia/model/SearchForFacetValuesResponseFacetHits.java
package com.algolia.model;

import com.google.gson.annotations.SerializedName;
import java.util.Objects;

/** SearchForFacetValuesResponseFacetHits */
public class SearchForFacetValuesResponseFacetHits {

  @SerializedName("value")
  private String value;

  @SerializedName("highlighted")
  private String highlighted;

  @SerializedName("count")
  private Integer count;

  public SearchForFacetValuesResponseFacetHits value(String value) {
    this.value = value;
    return this;
  }

  /**
   * Raw value of the facet.
   *
   * @return value
   */
  @javax.annotation.Nonnull
  public String getValue() {
    return value;
  }

  public void setValue(String value) {
    this.value = value;
  }

  public SearchForFacetValuesResponseFacetHits highlighted(String highlighted) {
    this.highlighted = highlighted;
    return this;
  }

  /**
   * Markup text with occurrences highlighted.
   *
   * @return highlighted
   */
  @javax.annotation.Nonnull
  public String getHighlighted() {
    return highlighted;
  }

  public void setHighlighted(String highlighted) {
    this.highlighted = highlighted;
  }

  public SearchForFacetValuesResponseFacetHits count(Integer count) {
    this.count = count;
    return this;
  }

  /**
   * How many objects contain this facet value. This takes into account the extra search parameters
   * specified in the query. Like for a regular search query, the counts may not be exhaustive.
   *
   * @return count
   */
  @javax.annotation.Nonnull
  public Integer getCount() {
    return count;
  }

  public void setCount(Integer count) {
    this.count = count;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    SearchForFacetValuesResponseFacetHits searchForFacetValuesResponseFacetHits = (SearchForFacetValuesResponseFacetHits) o;
    return (
      Objects.equals(this.value, searchForFacetValuesResponseFacetHits.value) &&
      Objects.equals(
        this.highlighted,
        searchForFacetValuesResponseFacetHits.highlighted
      ) &&
      Objects.equals(this.count, searchForFacetValuesResponseFacetHits.count)
    );
  }

  @Override
  public int hashCode() {
    return Objects.hash(value, highlighted, count);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class SearchForFacetValuesResponseFacetHits {\n");
    sb.append("    value: ").append(toIndentedString(value)).append("\n");
    sb
      .append("    highlighted: ")
      .append(toIndentedString(highlighted))
      .append("\n");
    sb.append("    count: ").append(toIndentedString(count)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: clients/algoliasearch-client-php/lib/Model/Search/SearchForFacetValuesResponseFacetHits.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * SearchForFacetValuesResponseFacetHits Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class SearchForFacetValuesResponseFacetHits extends \Algolia\AlgoliaSearch\Model\AbstractModel implements ModelInterface, \ArrayAccess, \JsonSerializable
{
    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelTypes = [
        'value' => 'string',
        'highlighted' => 'string',
        'count' => 'int',
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $modelFormats = [
        'value' => null,
        'highlighted' => null,
        'count' => null,
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelTypes()
    {
        return self::$modelTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelFormats()
    {
        return self::$modelFormats;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'value' => 'setValue',
        'highlighted' => 'setHighlighted',
        'count' => 'setCount',
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'value' => 'getValue',
        'highlighted' => 'getHighlighted',
        'count' => 'getCount',
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     */
    public function __construct(array $data = null)
    {
        if (isset($data['value'])) {
            $this->container['value'] = $data['value'];
        }
        if (isset($data['highlighted'])) {
            $this->container['highlighted'] = $data['highlighted'];
        }
        if (isset($data['count'])) {
            $this->container['count'] = $data['count'];
        }
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        if (!isset($this->container['value']) || $this->container['value'] === null) {
            $invalidProperties[] = "'value' can't be null";
        }
        if (!isset($this->container['highlighted']) || $this->container['highlighted'] === null) {
            $invalidProperties[] = "'highlighted' can't be null";
        }
        if (!isset($this->container['count']) || $this->container['count'] === null) {
            $invalidProperties[] = "'count' can't be null";
        }

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    /**
     * Gets value
     *
     * @return string
     */
    public function getValue()
    {
        return $this->container['value'] ?? null;
    }

    /**
     * Sets value
     *
     * @param string $value raw value of the facet
     *
     * @return self
     */
    public function setValue($value)
    {
        $this->container['value'] = $value;

        return $this;
    }

    /**
     * Gets highlighted
     *
     * @return string
     */
    public function getHighlighted()
    {
        return $this->container['highlighted'] ?? null;
    }

    /**
     * Sets highlighted
     *
     * @param string $highlighted markup text with occurrences highlighted
     *
     * @return self
     */
    public function setHighlighted($highlighted)
    {
        $this->container['highlighted'] = $highlighted;

        return $this;
    }

    /**
     * Gets count
     *
     * @return int
     */
    public function getCount()
    {
        return $this->container['count'] ?? null;
    }

    /**
     * Sets count
     *
     * @param int $count How many objects contain this facet value. This takes into account the extra search parameters specified in the query. Like for a regular search query, the counts may not be exhaustive.
     *
     * @return self
     */
    public function setCount($count)
    {
        $this->container['count'] = $count;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param int $offset Offset
     *
     * @return bool
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param int $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param int $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }
}

